src/buffer.rs

```diff
@@ -768,6 +768,29 @@ where
     }
 }
 
+impl<P, Container> ImageBuffer<P, Container>
+where
+    P: Pixel<Subpixel = u8> + 'static,
+    Container: Deref<Target = [u8]>,
+{
+    /// Saves the buffer to a file at the specified path in
+    /// the specified format.
+    pub fn save_with_format<Q>(&self, path: Q, format: ImageFormat) -> io::Result<()>
```

Comment:
Can this doc get one sentence linking to `save_buffer_with_format` which contains the extensive documentation about supported images?